File: 卡码/代码随想录/1/08-贪心算法/Js/03-困难/LeetCode_0452.js
//452. 用最少数量的箭引爆气球
const findMinArrowShots = function(points) {
	if(points.length==0) return 0;
	points.sort((p1,p2)=>p1[0]-p2[0]);
	let arrow=1;
	for(let i=1;i<points.length;i++){
		if(points[i][0]>points[i-1][1]) arrow++;
		else points[i][1]=Math.min(points[i][1],points[i-1][1]);
	}
	return arrow;
};

File: 卡码/代码随想录/1/08-贪心算法/Go/03-困难/LeetCode_0452.go
// 452. 用最少数量的箭引爆气球
package main

import (
	"math"
	"sort"
)

func findMinArrowShots(points [][]int) int {
	if len(points)==0 {
		return 0
	}
	arrow:=1
	sort.Slice(points,func (i,j int) bool{
		return points[i][0]<points[j][0]
	})
	for i:=1;i<len(points);i++ {
		if points[i][0]>points[i-1][1] {
			arrow++
		}else{
			points[i][1]=int(math.Min(float64(points[i][1]),float64(points[i-1][1])))
		}
	}
	return arrow
}
